src/utils/CalendarUtils/Render/Render.js

```diff
@@ -10,3 +12,16 @@ export const calendarWithContributions = (container, calendar, totalContribution
   calendarContainer.innerHTML = stringifiedHTMLContent;
   calendarContainer.prepend(calendarHeader);
 };
+
+export const defaultUserCalendar = async (state) => {
+  const defaultUserJsonCalendar = await GitHub.getJsonFormattedCalendarSync(
+    state.configs.proxyServerUrl, DefaultUsers.GitHub,
+  );
+
+  const restoredDefaultUserCalendar = GitHub.restoreCalendarValues(defaultUserJsonCalendar);
+
+  state.setStateAndRender({
```

Comment:
State**And**Render sounds like an indication that these could be separated :)

Reply:
Actually, the `render` is separated, but you're right.

Btw, my concern is that after each `setState` call we'd like to re-render the whole calendar. As a result, the same lines of codes will be written similarly to this example:

```
state.setState({
   updatedActualCalendar: BasicCalendar,
   currentUserTotalContributions: 0,
});

CalendarUtils.Render.calendarWithContributions(state);
```

I suspect that it's an indication to reconsider the current structure. How about creating a `Calendar`/`CalendarDetails` - instead of `State` as the naming can be misleading - class that holds:
- the whole logic of rendering the actual calendar according to the state of the fields within the class - and also able to establish the basic appearance(basic calendar, default user calendar),
- after each data-update, it re-renders the screen by calling its own renderer.

Example:
```
export default class Calendar {
   constructor(...) {
      // receives the "configs" and sets the default values as before
   }

   renderActualCalendar() {
      // ...
   }

   async renderBasicAppearance() {
      // renders BasicCalendar and the default user calendar synchronously
   }

   updateCalendarDetails(data) {
      // update values, call renderActualCalendar - or run it manually after each data-update
   }
};
```

It can even expose other functions later.

Reply:
I like that, but i'd still keep the State separately and use accessors.